docs/source/our_stack.rst

```diff
@@ -10,31 +10,104 @@ The most up-to-date reference on our codebase is our code itself.
+assigns all robots a task based on some coordination. Our
+system utilizes a play selector that chooses a play based on world state. After
+that play assigns roles, each role selects a tactic based on their progress, and
+tactic select an atomic skill to perform. The gameplay loop is set to tick at
+60Hz. The code block below describes this process. Each time the timer
+ends, the callback method is entered:
 
-ROS2
-----
+.. code-block:: python
 
-Vision
-------
+        timer_period = 1 / 60  # seconds
+        self.timer = self.create_timer(timer_period, self.gameplay_tick)
 
-Gameplay
---------
+Motion Control
+--------------
+A lot of the motion control is done on robot by the firmware, but part of the
+motion control is handled by software. See the ``control`` directory,
+``motion_control.cpp`` uses PID to correct robot position. The PID
+controller implementation is not in software, but it is in the
+`fshare repository <https://github.com/RoboJackets/robocup-fshare>`_.
+Correction of robot velocity is done by robot firmware (the code that runs on
+each robot's microcontroller) in the separate firmware repository.
 
 Motion Planning
 ---------------
+Motion planning is about generating a path from a robot's current position and
+orientation (pose) to a goal supplied by gameplay. Starting at the lowest
+level, we use `RRT <https://www.youtube.com/watch?v=Ob3BIJkQJEw>`, or Rapidly exploring Random Trees, for path generation. When the straight line path to a goal fails, we utilize the path generated by RRT (if there are no
+obstacles in that straight line why bother with RRT). We have our own RRT
+implementation in `a separate repository <https://github
+.com/RoboJackets/rrt>`_ which we use as a submodule. This
+way others can easily use our RRT implementation if they want.
+
+But we can't send a path to a robot and expect it to figure it out. We need
+to give our firmware a single velocity command for that instant of planning.
+The file ``soccer/src/soccer/planning/primitives/velocity_profiling.cpp``
+finds that velocity trajectory which we can then return as the current
+trajectory as the robot.
+
+At the highest level of planning, we have several different planners which the
+planning_node selects based on the skill that gameplay requested. They
+have different behaviors and based on the their names which give you insight on their purpose. They are all located
+in ``soccer/src/soccer/planning/planner``.
+
+Although this is not an extensive explanation of everything planning does, it is a good start, and looking
+through the ``planning`` folder will give you more information about this topic.
+
+Processor
+---------
+It is defined in ``soccer/src/soccer/processor.cpp``, and it
+starts multiple temp nodes. It then updates the context in an infinite loop. Context is used to update the ui. It is good that most other nodes
+are not dependent on this loop.
 
 Radio
 -----
+See the ``/radio`` directory. We have a sim radio for when using virtual
```

Comment:
This section is a little weirdly structured. It would be clearer imo if it was:
* We have a sim radio and a real radio
* Sim radio does this
* Real radio does this